src/test/java/com/github/emilshina/testcases/Data.java

```diff
@@ -0,0 +1,22 @@
+package com.github.emilshina.testcases;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Custom annotation for test data.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface Data {
+
+    String source() default "";
+
+    String dataType() default "";
```

Comment:
it makes no sense to pass type through annotation, as you already have a file extension and overridden interface method... let's assume that you'll work only with files...